File: 2021/day-02/day02-1.js
const fs = require('fs');

let horizontal_position = 0;
let depth = 0;
const datas = fs.readFileSync('input.txt', 'utf-8').split(/\n/).map(line => {
    let results = line.split(' ');
    return { direction: results[0], value: parseInt(results[1]) };
});
datas.map( data => {
    switch(data['direction']) {
        case 'forward':
            horizontal_position += data['value'];
            break;

        case 'up':
            depth -= data['value'];
            break;

        case 'down':
            depth += data['value'];
            break;
    }
});
console.log(horizontal_position * depth);

File: 2021/day-02/day02-1.py
import re

file = open('input.txt', 'r')
lines = file.readlines()

horizontal_position = 0
depth = 0
for line in lines:
    m = re.match("([a-z]*)\s([0-9]*)", line)
    direction = m.groups()[0]
    value = int(m.groups()[1])
    if direction == 'forward':
        horizontal_position += value
    elif direction == 'up':
        depth -= value
    elif direction == 'down':
        depth += value

print(horizontal_position * depth)
